README.md

```diff
@@ -1,16 +1,42 @@
 wikidata-stuff  [![Build Status](https://travis-ci.org/lokal-profil/wikidata-stuff.svg?branch=master)](https://travis-ci.org/lokal-profil/wikidata-stuff)
 ==============
 
-Random scripts used for Wikidata. These require [pywikibot](https://github.com/wikimedia/pywikibot-core)
+Random scripts used for doing mass imports to Wikidata. The emphasis is on sourcing
+Claims whether new or pre-existing. As such any imported statement will follow the
+following decision tree (any added statement includes the source):
 
-* WikidataStuff.py: A set of generally useful functions for interacting
-with Wikidata using pywikibot.
-* WikidataStringSearch.py: A database hookup (to be run from labs) for
+* The property has not yet been used. Statement is added
+* The property is already used and:
+   * The value is different. An new statement is added.
+   * The value is the same but sources differ. The new source is added to the
+     existing claim.
+   * The value is the same and (relevant parts of) the source are the same. No
+     information is added.
+
+For details on how qualifiers are handled, see `WikidataStuff.WikidataStuff.match_claim()`.
+
+For details on how sources are compared, see `WikidataStuff.Reference`.
+
+## Components
+
+* `WikidataStuff.py`:
+  * A set of generally useful functions for interacting with Wikidata using pywikibot.
+  * `Reference`: A class representing the source claims.
+  * `Qualifier`: A class representing qualifier claims.
+  * `Statement`: A class representing a statement (i.e. value, qualifiers and references).
+* `WikidataStringSearch.py`: A database hookup (to be run from labs) for
```

Comment:
I think 'labs' as a name is being phased out (https://phabricator.wikimedia.org/phame/post/view/59/labs_and_tool_labs_being_renamed/), so since we're updating this document anyway, might be good to look for these before someone complains.

Reply:
Good catch. Changed to "Toolforge"